### PR TITLE
Fix csv.status.RotatedAt update

### PR DESCRIPTION
**Pull request by perdasilva**
Signed-off-by: Per Goncalves da Silva <pegoncal@redhat.com>

**Description of the change:**
This PR updates the CSV InstallReady state to only update the`csv.status.CertsLastUpdatedAt` and `csv.status.CertsRotateAt`
status fields if they are not set (and the csv contains CAResources), or if expired.

NOTE: I couldn't find a clear way to surface the certificate expiry and rotation dates from the installer. So, I patched and cleaned the current implementation. Though, if anyone has good ideas, I'm all ears!

**Motivation for the change:**
The complete context can be seen in the [bz](https://bugzilla.redhat.com/show_bug.cgi?id=2020484).
Essentially, if the package server pod get killed, the csv moves to a `Failed` states and progresses forward to `Succeeded`. During the `InstallReady` phase, the `installer` would not rotate the csv if the certificate has not expired yet. BUT, the csv status fields were still being updated, moving the rotatedAt out another 2 years. This creates skew between the certificate's rotation time and the what's in the csv status. However, the operator decides when to rotate the certificate based on the `status.RotateAt`. Which would in turn push out the rotation erroneously.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky
- [ ] Tests that remove the `[FLAKE]` tag are no longer flaky

### CODE REVIEW

File: pkg/controller/operators/olm/operator.go [hunk 1]
```diff
@@ -1888,10 +1888,17 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 
-		if out.HasCAResources() {
+		// Only update certificate status if:
+		//  - there the CSV has CAResources; and
+		//  - it has not already been set, or they should be rotated
+		// See bug: https://bugzilla.redhat.com/show_bug.cgi?id=2020484 for more information
```
Comment (awgreene) [inline pkg/controller/operators/olm/operator.go:1894]:
```suggestion
		// Only update certificate status if:
		//  - the CSV has CAResources; and
		//  - it has not already been set, or they should be rotated
```

Comment (awgreene) [reply pkg/controller/operators/olm/operator.go:1894]: +1 IRT dropping the BZ reference.

Comment (awgreene) [reply pkg/controller/operators/olm/operator.go:1894]: BZ is captured in the Pull Request description anyways.

**Review by awgreene — APPROVED** [review]
Looks good, giving this PR my approval.
/approve

File: pkg/controller/operators/olm/operator.go (hunk 1 above)
Comment (perdasilva) [reply pkg/controller/operators/olm/operator.go:1894]: Lol - just wrote that in the previous response. I'm good with removing that.

**Comment by dinhxuanvu** [comment]
I can lgtm after test case(s) is added and rebased.

**Review by tylerslaton — APPROVED** [review]
Great work here Per, giving it a rubber stamp and echoing what everyone else has said. The implementation looks good to me and I'll give it the official label once the test cases come through.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by dinhxuanvu** [comment]
I think the PR looks good. Ready for merge whenever you remove the WIP tag on the title.

**Pull request merged**